coinbaseadvanced/client.py

```diff
@@ -557,6 +594,43 @@ def get_product_candles(
         product_candles = CandlesPage.from_response(response)
         return product_candles
 
+    def get_product_candles_all(
+           self,
+            product_id: str,
+            start_date: datetime,
+            end_date: datetime,
+            granularity: Granularity) -> CandlesPage:
+        """
+        Gets all requested product candles
+        """
+
+        # pre-calculate granularity entries in minutes 
+        granularity_minutes = timedelta(minutes=GRANULARITY_MAP_IN_MINUTES[granularity.value])
+        # request size of 300 (max allowed by coinbase)
+        granularity_minutes_x300 = granularity_minutes * 300
+
+        product_candles = CandlesPage({})
+
+        # run through from most recent to oldest to preserve time order in list
+
+        end = end_date
+        # calculate start date
+        begin = end_date - granularity_minutes_x300
+
+        # while we still have not gotten all the requested candles loop until all are requested
+        while end > start_date:
+            # avoid asking for more than requested
+            if begin < start_date:
+                begin = start_date
+            # get the next batch and extend the list
+            product_candles.candles.extend(self.get_product_candles(product_id, begin, end,granularity).candles)
+            # offset end by one granularity to avoid duplicates
+            end = begin - granularity_minutes
+            # recalculate start for the previous (older) 300 candles
+            begin = begin - (granularity_minutes_x300 + granularity_minutes)
+        
+        return product_candles
+
```

Comment:
Looks good. Do you want to add a unit test, or do you want me to help with that before merging? We can also add the unit tests post-merging (before launching `0.2.0` release).

Reply:
I've looked into the existing unit tests, but I don't yet grok them. I'm not certain what is actually being tested.

I would appreciate you helping on the tests until I figure them out.